addons/cert.py

```diff
@@ -234,6 +234,31 @@ def msc30(data):
         if simpleMatch(token, "rand ( )") and isStandardFunction(token):
             reportError(token, 'style', 'Do not use the rand() function for generating pseudorandom numbers', 'MSC30-c')
 
+# MSC24-C
+# Do not use deprecated or obsolescent functions
+def msc24(data):
+    for token in data.tokenlist:
+        if token.str == 'asctime':
```

Comment:
It looks pretty good. I just think we can try to make this check a bit more safe. Check that next token is a "(" and that this token is astOperand1..

    if token.next is None or token.next.str != "(" or token != token.next.astOperand1:
        continue

Reply:
An additional heuristic would be to check that the number of parameters in the function argument matches. For instance if there is a atof function call without parameters then no warning should be written. But I'll merge this now. Please feel free to write this heuristic later.

Reply:
`isFunctionCall` has an optional parameter `number_of_arguments` with which that could be verified.
The condition `if not isFunctionCall(...` would have to be removed then and instead of comparing `token.str` to the function name here, it should be possible to use something like `if isFunctionCall(token, ('atoi'), 1):`

Reply:
yes I agree with this. It would feel better if the number of arguments is checked.

Reply:
hmm I don't get it: if i change the code as you mentioned e.g. 
        if isFunctionCall(token, ('asctime'), 1):
            reportError(token,'style','Do no use asctime() better use asctime_s()', 'MSC24-C')

the check will also report a warning for this line:
    time ( &rawtime );
I thought that searching in a list/array with 'in' does only give you exact matches of the string you are searching for?

Reply:
I thought the same and would not have expected that it finds substrings.
It seems that tuples are the problem here.
Read an interesting detail here:
http://www.openbookproject.net/books/bpp4awd/ch03.html
If you specify a tuple with only one element then a comma must be added after this element. Otherwise python seems to ignore the brackets.
Sorry, i didn't know that. I'm still somewhat new to python. 
`if isFunctionCall(token, ('asctime',), 1):` should work i guess.

Reply:
No biggy. I'm also new to python so I also have to fight with some python querks here and there :)
However I pushed the changes to the branch, it's working now ;)